src/core/core.scale.js

```diff
@@ -1017,18 +1017,15 @@ var Scale = Element.extend({
 				continue;
 			}
 
-			if (i === me.zeroLineIndex && options.offset === offsetGridLines) {
-				// Draw the first index specially
-				lineWidth = gridLines.zeroLineWidth;
-				lineColor = gridLines.zeroLineColor;
-				borderDash = gridLines.zeroLineBorderDash || [];
-				borderDashOffset = gridLines.zeroLineBorderDashOffset || 0.0;
-			} else {
-				lineWidth = valueAtIndexOrDefault(gridLines.lineWidth, i, 1);
-				lineColor = valueAtIndexOrDefault(gridLines.color, i, 'rgba(0,0,0,0.1)');
-				borderDash = gridLines.borderDash || [];
-				borderDashOffset = gridLines.borderDashOffset || 0.0;
-			}
+			context = {
+				scale: me,
+				tick,
+			};
+
+			lineWidth = resolve([gridLines.lineWidth, 1], context, i);
+			lineColor = resolve([gridLines.color, 'rgba(0,0,0,0.1)'], context, i);
+			borderDash = gridLines.borderDash || [];
+			borderDashOffset = resolve([gridLines.borderDashOffset, 0], context, i);
```

Comment:
One thing to note about `i` here - the user won't know in advance how many ticks there will be if `autoSkip` is enabled. Do we want to pass `i` still or maybe only if `autoSkip` is disabled?  I'm also curious whether we want to keep indexable options around since they're less powerful than scriptable options and predate them

Reply:
The index options can work well, with less memory, if one specifies `min`, `max`, and `stepSize`. For simplicity I'd keep supporting them